src/VisualStudio/Core/Def/Progression/GraphNavigatorExtension.cs

```diff
@@ -77,17 +80,10 @@ public void NavigateTo(GraphObject graphObject)
             // Notify of navigation so third parties can intercept the navigation
             if (symbolId != null)
             {
-                var symbolNavigationService = _workspace.Services.GetService<ISymbolNavigationService>();
                 var symbol = symbolId.Value.Resolve(await project.GetCompilationAsync(cancellationToken).ConfigureAwait(false), cancellationToken: cancellationToken).Symbol;
-
-                // Do not allow third party navigation to types or constructors
-                if (symbol != null &&
-                    symbol is not ITypeSymbol &&
-                    !symbol.IsConstructor() &&
-                    await symbolNavigationService.TrySymbolNavigationNotifyAsync(symbol, project, cancellationToken).ConfigureAwait(false))
```

Comment:
all of this logic is already embedded in GoToDefinitionHelpers.GetDefinitionLocationAsync.  The goal is to get everything on that instead of having so many places reimplement all this logic.